interp/type.go

```diff
@@ -1246,21 +1246,19 @@ func (t *itype) lookupBinMethod(name string) (m reflect.Method, index []int, isP
 	if t.cat == ptrT {
 		return t.val.lookupBinMethod(name)
 	}
+	for i, f := range t.field {
+		if f.embed {
+			if m2, index2, isPtr2, ok2 := f.typ.lookupBinMethod(name); ok2 {
+				index = append([]int{i}, index2...)
+				return m2, index, isPtr2, ok2
+			}
+		}
+	}
 	m, ok = t.TypeOf().MethodByName(name)
 	if !ok {
 		m, ok = reflect.PtrTo(t.TypeOf()).MethodByName(name)
 		isPtr = ok
 	}
-	if !ok {
```

Comment:
If above L1254, we just do:

```
	if ok {
		index = []int{m.Index}
	}
```
then the added test works too. So what is wrong with that solution?

Reply:
answered by chat: it works by coincidence (m.Index is 0 too in this case), but m.Index does not have the meaning we want (i.e. it is not the position within the struct of the field that implements the method).